src/org/labkey/targetedms/folderImport/PanoramaQCSettings.java

```diff
@@ -0,0 +1,489 @@
+                }
+            },
+    QC_ANNOTATION_TYPE (TargetedMSSchema.TABLE_QC_ANNOTATION_TYPE, QCFolderConstants.QC_ANNOTATION_TYPE)
+            {
+                @Override
+                public void writeSettings(VirtualFile vf, Container container, User user) throws Exception
+                {
+                    TargetedMSSchema schema = new TargetedMSSchema(user, container);
+                    ContainerFilter cf = ContainerFilter.getContainerFilterByName(ContainerFilter.Type.CurrentPlusProjectAndShared.name(), container, user);
+                    TableInfo ti = schema.getTable(getTableName(), cf);
+                    assert ti != null;
+                    List<ColumnInfo> userEditableCols = ti.getColumns().stream().filter(ci -> ci.isUserEditable()).collect(Collectors.toList());
+                    ResultsFactory factory = ()-> QueryService.get().select(ti, userEditableCols, null, null);
+                    writeSettingsToTSV(vf, factory, getSettingsFileName(), getTableName());
+                }
+
+                @Override
+                public int importSettingsFromFile(FolderImportContext ctx, VirtualFile panoramaQCDir, @Nullable TargetedMSSchema schema, @Nullable TableInfo ti, @Nullable QueryUpdateService qus, @Nullable BatchValidationException errors) throws Exception
+                {
+                    assert schema != null;
+                    TableInfo tinfo = schema.getTable(getTableName(), ContainerFilter.getContainerFilterByName(ContainerFilter.Type.CurrentPlusProjectAndShared.name(), ctx.getContainer(), ctx.getUser()));
+                    return importSettings(ctx, tinfo, getSettingsFileName(), panoramaQCDir, errors, qus);
+                }
+            },
+    QC_ANNOTATION (TargetedMSSchema.TABLE_QC_ANNOTATION, QCFolderConstants.QC_ANNOTATION_FILE_NAME)
+            {
+                @Override
+                public void writeSettings(VirtualFile vf, Container container, User user) throws Exception
+                {
+                    TargetedMSSchema schema = new TargetedMSSchema(user, container);
+                    TableInfo ti = schema.getTable(getTableName());
+                    assert ti != null;
+                    SQLFragment sql = new SQLFragment("SELECT qcAnnotationType.Name AS QCAnnotationTypeId, qcAnnotation.Description, qcAnnotation.Date ")
+                            .append(" FROM qcAnnotation")
+                            .append(" INNER JOIN qcAnnotationType")
+                            .append(" ON qcAnnotation.QCAnnotationTypeId = qcAnnotationType.Id");
+
+                    Map<String, TableInfo> tableMap = new HashMap<>();
+                    tableMap.put("qcAnnotation", ti);
+                    tableMap.put("qcAnnotationType", schema.getTable(TargetedMSSchema.TABLE_QC_ANNOTATION_TYPE));
+
+                    ResultsFactory factory = ()-> QueryService.get().selectResults(schema, sql.getSQL(), tableMap, null, true, true);
```

Comment:
Why do we need to pass in the tableMap here? Can't QueryService resolve the tables itself from the schema?